azurelinuxagent/ga/monitor.py

```diff
@@ -279,11 +279,17 @@ def daemon(self):
 
     def add_sysinfo(self, event):
         sysinfo_names = [v.name for v in self.sysinfo]
+        copy_param = []
+
         for param in event.parameters:
-            if param.name in sysinfo_names:
-                logger.verbose("Remove existing event parameter: [{0}:{1}]", param.name, param.value)
-                event.parameters.remove(param)
-        event.parameters.extend(self.sysinfo)
+            if param.name not in sysinfo_names:
+                logger.verbose("Adding existing event parameter: [{0}:{1}]", param.name, param.value)
+                copy_param.append(param)
+            else:
+                logger.verbose("Removing existing event parameter: [{0}:{1}]", param.name, param.value)
+
+        copy_param.extend(self.sysinfo)
+        event.parameters = copy_param
 
```

Comment:
The earlier logic doesn't cleanly remove items from the list. When the fields to be removed come next to each other in the list, the list.remove would stip the second occurrence of the item to be removed.